ccmlib/cluster.py

```diff
@@ -247,8 +247,16 @@ def add(self, node, is_seed, data_center=None):
         for trace_class in self._trace:
             node.set_log_level("TRACE", trace_class)
 
+        if data_center is None:
```

Comment:
if I understand correctly the logic here is:
"if data center is not set for then newly added node should get DC from first DC used in existing cluster nodes"

so we can move L242 after calculation of initial `data_center`
```
if data_center is None:
    cluster_dcs = [n.data_center for n in self.nodelist() if n.data_center is not None]
    data_center = cluster_dcs[0] if cluster_dcs else None
node.data_center = data_center
```

Reply:
I wonder if it is possible that some test / usage would like to use node without DC being set?
and in such case using `None` will not work, as it will be ignored.

Reply:
I have another doubt:
if there are different DCs used in the cluster, and someone adds... it will get first DC.

so maybe instead of quessing / overriding we should assert that DC is provided if there is more than 1 dc in the cluster?

In such case we can assert `if data_center is None:`
```
assert len(set(cluster_dcs)) <= 1, "..." 
```

Reply:
DC can be None when there are no DC's defined and it is fine (`SimpleStrategy`). Everybody jumps onto the 'default' one.

On NTPS a DC of None will have the node join the default DC so we need to account for that. I went for the default one when `None` instead of relying on the list of nodes as I don't have ordering guarantees afaik on the list.

I don't think a multi-DC setup with respecting 'None' makes sense as it breaks topology which is exactly what we are fixing here. Hope it makes sense.

Reply:
I have updated the proposal to reflect the fact that we set to "dc1" explicitly,
assuming that any new node will join first dc
and first dc in ccm is dc1

Reply:
if we want to reflect "always use dc1 if any dc is used" then
```
if data_center is None:
    cluster_dcs = [n.data_center for n in self.nodelist() if n.data_center is not None]
    data_center = "dc1" if cluster_dcs else None
node.data_center = data_center
```

Reply:
Erroring out on missing DC when there were multiple DCs present was my first proposal. But then I noticed ccm forces dc1 and that is the default one as well. So I assumed it safe to use dc1 directly.

Some form of `len(nodelist().data_center) !=0` should work to the same effect. Anyway I say we wait for further reviews before pushing changes to bulk change in one go.

Reply:
[Here](https://app.circleci.com/pipelines/github/bereng/cassandra/199/workflows/dbcb0f6e-1bb0-4334-b60c-07f07cb71792/jobs/1741) is the run where I would fail if no dc was specified. There were like 20-ish failures but it would be easy to fix those if we prefer that option.

Reply:
I think it is better to explicitly say in a multidc scenario which dc the node is added to
https://www.python.org/dev/peps/pep-0020/#id2

Reply:
@bereng, I saw only three tests failing, is this the right link? 
My personal preference - do it as close as possible to the reality and any implications or possible second guesses/confusion - commented + warnings logged so that people use this correctly and have the right expectations

Reply:
Ok switching to erroring out if DC is missing.